src/services/deduplication/backup.py

```
@@ -30,6 +34,41 @@
+
+    Verifies the target is a regular file (not a symlink) via lstat before
+    unlinking.  SafeDir.lstat uses follow_symlinks=False so it stats the
+    entry itself; S_ISREG catches symlinks and any other non-regular type.
+    Returns True if the file was removed, False otherwise.
+
+    On Windows (no SafeDir support) falls back to plain unlink.
+    """
+    if sys.platform == "win32":
+        try:
+            path.unlink()
+            return True
```

Ruff (0.15.13)

[warning] Consider moving this statement to an `else` block

(TRY300)